### ISSUE TITLE
False-positive D214/405/410/411

### ISSUE THREAD

**Issue by twoertwein** [charliermarsh/ruff#1561]
```py
from typing import Any

def get_object(parent: object, attributes: list[str]) -> Any:
    """Traverse an object to get a nested attribute.

    Args:
        parent:
            Object to be traversed.
        attributes:
            Names of attributes.

    Returns:
        The nested object.
    """
    ...
```

Ruff reports the following for the above code (`convention = "google"` in pyproject.toml):

> test.py:5:5: D410 Missing blank line after section ("Args")
> test.py:5:5: D405 Section name should be properly capitalized ("attributes")
> test.py:5:5: D214 Section is over-indented ("attributes")
> test.py:5:5: D411 Missing blank line before section ("attributes")

and transforms it to

```py
from typing import Any

def get_object(parent: object, attributes: list[str]) -> Any:
    """Traverse an object to get a nested attribute.

    Args:
        parent:
            Object to be traversed.

    Attributes:
            Names of attributes.

    Returns:
        The nested object.
    """
    ...
```

**Comment by charliermarsh** [comment]
Hah yeah that’s tough to fix in the general case. It’s hard to know that attributes is an actual parameter there and not an Attributed section (which is a valid Google docstring section) with improper spacing.

I think it would fix if you did “attributes (list[str]): …” as a workaround?

**Comment by charliermarsh** [comment]
Either of these will work:

```py
def get_object(parent: object, attributes: list[str]) -> Any:
    """Traverse an object to get a nested attribute.

    Args:
        parent: Object to be traversed.
        attributes: Names of attributes.

    Returns:
        The nested object.
    """
    ...
```

Or:

```py
def get_object(parent: object, attributes: list[str]) -> Any:
    """Traverse an object to get a nested attribute.

    Args:
        parent (object):
            Object to be traversed.
        attributes (list[str]):
            Names of attributes.

    Returns:
        The nested object.
    """
    ...
```
